tests/cpuid_test.c

```diff
@@ -24,6 +24,7 @@ static int s_cpuid_test_fn(struct aws_allocator *allocator, void *ctx) {
     aws_cpu_has_feature(AWS_CPU_FEATURE_SSE_4_2);
     aws_cpu_has_feature(AWS_CPU_FEATURE_AVX2);
     aws_cpu_has_feature(AWS_CPU_FEATURE_ARM_CRC);
```

Comment:
should we be asserting in this test in any way? how do we know it works?

Reply:
We cannot assert on it, it has different result on different platform...

Just run them and not crash?

Reply:
how do we know if we are actually picking up the feature on the platforms where its supported? wasnt the original issue because of incorrect check. 
Does not have to be addressed in this pr, but something we need to think about